### PR TITLE
Add sort-option, fix key for selected, and fix double submit

### PR DESCRIPTION
**Pull request by nathanvda**
> As mentioned in the the PR default text, I targeted the "experimental" branch!

So maybe this is not ideal, but the first two commits were discussed in #167 (and #176) where I 

- make sure the key is used when setting a selected value
- make sure the sorting is optional 

I set the default option to not-sorted, which may not be your "default" choice. That could easily by changed? I added tests for these two cases. 

Furthermore I added a naive approach to prevent double submits. I noticed in my code that every form.submit was called twice. I was not able to get to the real cause of that, but did manage a simple work-around preventing it.

I hope this is useful. If you have any feedback, let me know.

### CODE REVIEW

**Comment by nathanvda** [comment]
Sorry, it got a bit dirty when fixing the codacy reviews. Squash merge it?

File: src/jquery.jeditable.js [hunk 1]
```diff
@@ -668,13 +677,14 @@ var _supportInType = function (type) {
...
-                    });
-
+                    if (settings.sortSelectOptions) {
+                        // sort it
+                        tuples.sort(function (a, b) {
+                            a = a[1];
+                            b = b[1];
+                            return a < b ? -1 : (a > b ? 1 : 0);
+                        });
+                    }
```
Comment (NicolasCARPi) [inline src/jquery.jeditable.js]: Trailing whitespace (add a plugin to your editor to see them!)

File: src/jquery.jeditable.js [hunk 2]
```diff
@@ -687,14 +697,14 @@ var _supportInType = function (type) {
...
                         if (key !== 'selected') {
                             option = $('<option />').val(key).append(value);
-                        }
 
-                        // add the selected prop if it's the same as original or if the key is 'selected'
-                        if (key === 'selected' || key === $.trim(original.revert)) {
-                            $(option).prop('selected', 'selected');
-                        }
+                            // add the selected prop if it's the same as original or if the key is 'selected'
+                            if (json['selected'] === key || key === $.trim(original.revert)) {
```
Comment (NicolasCARPi) [inline src/jquery.jeditable.js]: use dot notation (json.selected)

File: src/jquery.jeditable.js [hunk 3]
```diff
@@ -339,13 +340,21 @@
 
                 form.submit(function(e) {
 
+                    /* Do no submit. */
+                    e.preventDefault();
+                    e.stopPropagation();
+
+                    if (isSubmitting) {
```
Comment (NicolasCARPi) [inline src/jquery.jeditable.js]: I don't get this part. Can you explain here why you added the `isSubmitting` var?

Comment (nathanvda) [reply src/jquery.jeditable.js]: As I explained above: I experienced double submits, `form.submit` was called twice, and this was my work-around. You do not have to include this if I am the only person experiencing this? I have been experiencing this problem for a long time, but it didn't really bother me (my data was just saved twice) (I noticed this in the logfiles but just ignored it), until I attached a callback and then it became a "visual" problem. I tried making a test-case for it, but I need to find a way how to capture the calling of the ajax-request.

Comment (nathanvda) [reply src/jquery.jeditable.js]: To be clear: the user did not do anything strange, just press enter, or click 'Ok' (to start the submit action). It occurs in chrome, ie and firefox. The only thing deviating from the default is I style my buttons using bootstrap, which should not cause it imho?

**Comment by NicolasCARPi** [comment]
Hello, I have merged your contribution. I renamed the option in lower case to be consistant with the rest. And also removed the console warning.

Thanks :)

**Comment by nathanvda** [comment]
Thanks :clap: :clap: and it was my pleasure :+1: :+1: :wink:

**Pull request merged**